### ISSUE TITLE
The server returned a "405 Method Not Allowed"

### ISSUE THREAD

**Issue by TonyD3x** [MichaelBelgium/MV_Youtube#16]
![image](https://user-images.githubusercontent.com/29615495/157731558-4a337dd7-b27c-4c1f-bbd8-54fb55fce65a.png)

**Comment by MichaelBelgium** [comment]
What are you trying to do?

**Comment by TonyD3x** [comment]
I am using youtube play script in samp and it throws this web error: I am also using the API manually and it also throws the error

**Comment by MichaelBelgium** [comment]
It might call the url in a wrong way, that error comes up if the url only allows POST request but you're doing a GET request or vice-versa

**Comment by HebertLuiiz** [comment]
Hi, I'm having the same problem, I can't convert any links, but I can do research, could you leave a more detailed tutorial for us to create our api?

**Comment by MichaelBelgium** [comment]
@HebertLuiiz Is it 100% only when converting?

**Comment by HebertLuiiz** [comment]
I'm using your api, I haven't modified the include <MV_Youtube>, but I'm facing the same problem as TonyD3x, the api is not reading the link to convert, but it can read the link to do a search.

**Comment by MichaelBelgium** [comment]
Oh i see the issue already i think

Can you test it out if you change [HTTP_GET to HTTP_POST on this line](https://github.com/MichaelBelgium/MV_Youtube/blob/master/MV_Youtube.inc#L94) ?

**Comment by HebertLuiiz** [comment]
![Screenshot_20220506-171723](https://user-images.githubusercontent.com/105074040/167211017-b2e60b37-5aa8-4743-a931-d2d653b2f907.png)

http://youtube.michaelbelgium.me/api/converter/convert?url=https://youtu.be/f7JWDLFhR_c

**Comment by MichaelBelgium** [comment]
It only accepts POST requests, thats why it won't work in the browser just by going to it via url

To convert in browser you need to go to https://youtube.michaelbelgium.me/converter and use the form

**Comment by HebertLuiiz** [comment]
> Oh eu vejo o problema já eu acho
> 
> Você pode testá-lo se alterar [HTTP_GET para HTTP_POST nesta linha](https://github.com/MichaelBelgium/MV_Youtube/blob/master/MV_Youtube.inc#L94) ?

Yes but this line has an error in strmid, I'm trying to solve it but it seems to be right.

https://github.com/MichaelBelgium/MV_Youtube/blob/eae80b4cd913b1af3b5bbd3d1780bc38fe441866/MV_Youtube.inc#L89

**Comment by HebertLuiiz** [comment]
> Ele só aceita solicitações POST, por isso não funcionará no navegador apenas acessando via url
> 
> Para converter no navegador, você precisa acessar https://youtube.michaelbelgium.me/converter e usar o formulário

Using the form:
![image](https://user-images.githubusercontent.com/105074040/167212723-02c01960-15d7-4a9a-b1c5-91a718f525ac.png)

**Comment by MichaelBelgium** [comment]
What youtube url are you trying?


> 
> Yes but this line has an error in strmid, I'm trying to solve it but it seems to be right.
> 
> https://github.com/MichaelBelgium/MV_Youtube/blob/eae80b4cd913b1af3b5bbd3d1780bc38fe441866/MV_Youtube.inc#L89

Can't test it out right now but if you say HTTP_POST works, i can update the include

**Comment by HebertLuiiz** [comment]
> Qual url do youtube você está tentando?
> 
> Deixa pra lá eu acho https://youtu.be/f7JWDLFhR_c ?
> 
> > Sim mas esta linha tem um erro no strmid, estou tentando resolver mas parece estar certo.
> > https://github.com/MichaelBelgium/MV_Youtube/blob/eae80b4cd913b1af3b5bbd3d1780bc38fe441866/MV_Youtube.inc#L89
> 
> Que erro essa linha diz? Em geral, não posso testá-lo agora, mas se você disser que HTTP_POST funciona, posso atualizar o include

strmid error: error 035: MV_Youtube.inc(89) : error 035: argument type mismatch (argument 1)

If you can update the include it will help the community a lot, thanks for your attention

**Comment by MichaelBelgium** [comment]
* Using the form should be fixed now
* what if you remove the strmid line? I don't think it's needed anymore

**Comment by HebertLuiiz** [comment]
At the moment I have no way to access my computer, as soon as possible I will remove the line, perform the test with HTTP_GET and HTTP_POST and I will tell you the result.

**Comment by HebertLuiiz** [comment]
I used this link to perform a search and to play, both functions return the error 401, I took a random link from youtube, tested with the HTTP_GET and HTTP_POST functions.

https://www.youtube.com/watch?v=tUJXtHMXDY0

**Comment by MichaelBelgium** [comment]
401 means unauthorized, which also means the wrong or no token is passed on

**Comment by HebertLuiiz** [comment]
I put my token, and yes it works in the form

**Comment by MichaelBelgium** [comment]
Damn it, passing api_key via the url with a post request doesn't work... which means it won't work in samp, for now. I'm gonna think about a workaround

**Comment by MichaelBelgium** [comment]
@HebertLuiiz okay, I edited so the convert url can **accept POST and GET** requests. 
So the include doesn't need to be changed at all now. Except the if-strmid can be removed
Can you test it out? Undo everything you edited in the include but leave out the strmid line, which isn't needed

**Comment by HebertLuiiz** [comment]
It is returning error 405

**Comment by MichaelBelgium** [comment]
Hmm don't have that, it should work either way

**Comment by MichaelBelgium** [comment]
```
---------- Search response ----------
Title: Michael laat jury in verwarring achter door goocheltrucs - HOLLAhttps://youtube.com/watch?v=L6kKSOwd3PM
Link: https://youtube.com/watch?v=L6kKSOwd3PM
Title: Artist Spotlight: Micha�l Borremans ‼ A Complete and Chronologichttps://youtube.com/watch?v=sgBsB5M3k3E
Link: https://youtube.com/watch?v=sgBsB5M3k3E
Title: BBC&#39;s Great Continental Railway Journeys &quot;Amsterdam to https://youtube.com/watch?v=1hGpTHrE0LA
Link: https://youtube.com/watch?v=1hGpTHrE0LA
Title: Sorry Michael@Belgium :)
Link: https://youtube.com/watch?v=kjii0u6cslg
Title: My Name Is Michael (Belgium) - Thriller
Link: https://youtube.com/watch?v=tm_JB0szSUE

Video 0 started

---------- Testing video response 0 ----------
Title: Luis Fonsi, Demi Lovato - �chame La Culpa (Video Oficial)
Link: https://www.youtube.com/watch?v=TyHvyGVs42U
Duration: 210 seconds
Stream: https://youtube.michaelbelgium.me/storage/TyHvyGVs42U.mp3
Play for: 65535
Playlist id: -1

Added video 2 to playlist 0
Added video 1 to playlist 0

---------- Testing video response 1 ----------
Title: Jax Jones - Breathe (Official Video) ft. Ina Wroldsen
Link: https://www.youtube.com/watch?v=uGhKqb2Ow3E
Duration: 221 seconds
Stream: https://youtube.michaelbelgium.me/storage/uGhKqb2Ow3E.mp3
Play for: 65535
Playlist id: 0

---------- Testing video response 2 ----------
Title: Gabry Ponte, R3HAB, Timmy Trumpet - Call Me (Official Lyric Video)
Link: https://www.youtube.com/watch?v=NkrkAsRVLEA
Duration: 178 seconds
Stream: https://youtube.michaelbelgium.me/storage/NkrkAsRVLEA.mp3
Play for: 65535
Playlist id: 0

---------- Testing playlist response ----------
Id: 0
Name: Playlist 1
Songcount: 2
```

Tested out `test.pwn` with sampctl and everything is working. Do you also use sampctl or you installed the include manually?

**Comment by HebertLuiiz** [comment]
I use Include manually

**Comment by MichaelBelgium** [comment]
Did you use the new version?

**Comment by MichaelBelgium** [comment]
Always 405??
